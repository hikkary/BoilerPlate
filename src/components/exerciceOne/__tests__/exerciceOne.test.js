import React from 'react'
import ExerciceOne from '../'
import renderer from 'react-test-renderer'

import { shallow } from 'enzyme'

it('should match the snapshot (bad practice)', () => {
  const component = renderer
    .create(<ExerciceOne counterTitle="test" counterNumber={0} />)
    .toJSON()
  // `renderer`cree automatiquement un render en "profondeur"
  // changer un composant enfant peut faire rater ce test

  expect(component).toMatchSnapshot()
})

it('should match the snapshot (good Practice)', () => {
  const component = shallow(
    <ExerciceOne counterTitle="test" counterNumber={0} />
  )

  expect(component).toMatchSnapshot()
})

it('should have two buttons', () => {
  const component = shallow(
    <ExerciceOne counterTitle="test" counterNumber={0} />
  )
  const buttonElement = component.find('button')

  expect(buttonElement.length).toEqual(2)
})

it('should return null', () => {
  const component = shallow(
    <ExerciceOne counterTitle="test" counterNumber={0} disabled={true} />
  )

  // console.log(component.type())
  // permet de voir le type du composant, div si c'est une div.

  expect(component.type()).toBe(null)
})
