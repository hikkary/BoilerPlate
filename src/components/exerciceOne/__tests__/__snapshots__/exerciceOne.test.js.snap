// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot (bad practice) 1`] = `
<div
  className="counterContainer"
>
  <div>
    <span
      className="commonText"
    >
      test
    </span>
  </div>
  <div
    className="counterNumberContainer"
  >
    <span
      className="commonText"
    >
      0
    </span>
  </div>
  <div>
    <button
      className="styledButton"
      onClick={undefined}
    >
      <span
        className="commonText"
      >
        -
      </span>
    </button>
    <button
      className="styledButton"
      onClick={undefined}
    >
      <span
        className="commonText"
      >
        +
      </span>
    </button>
  </div>
</div>
`;

exports[`should match the snapshot (good Practice) 1`] = `
<div
  className="counterContainer"
>
  <div>
    <span
      className="commonText"
    >
      test
    </span>
  </div>
  <div
    className="counterNumberContainer"
  >
    <span
      className="commonText"
    >
      0
    </span>
  </div>
  <div>
    <button
      className="styledButton"
    >
      <span
        className="commonText"
      >
        -
      </span>
    </button>
    <button
      className="styledButton"
    >
      <span
        className="commonText"
      >
        +
      </span>
    </button>
  </div>
</div>
`;
